File: order-service/src/main/java/com/intexsoft/webshop/orderservice/producer/impl/OrderEventProducerImpl.java
package com.intexsoft.webshop.orderservice.producer.impl;

import com.intexsoft.webshop.messagecommon.event.order.OrderInitializedEvent;
import com.intexsoft.webshop.orderservice.producer.OrderEventProducer;
import com.intexsoft.webshop.orderservice.util.JsonUtils;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.amqp.core.TopicExchange;
import org.springframework.amqp.rabbit.core.RabbitTemplate;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
@Slf4j
public class OrderEventProducerImpl implements OrderEventProducer {
    private final RabbitTemplate rabbitTemplate;
    private final TopicExchange eventExchange;
    @Value("${rmq.event.order.routing-prefix}")
    private String routingPrefix;
    @Value("${rmq.event.order.routing-keys.order_initialized}")
    private String orderInitializedRoutingKey;

    @Override
    public void produceOrderInitializedEvent(OrderInitializedEvent orderInitializedEvent) {
        log.info("IN: produce {} message. The event message = {}",
                orderInitializedEvent.getClass().getName(), JsonUtils.getAsString(orderInitializedEvent));
        String routing = routingPrefix + orderInitializedRoutingKey;
        log.debug("Message is producing to exchange = {}, with routing = {}", eventExchange.getName(), routing);
        rabbitTemplate.convertAndSend(eventExchange.getName(), routing, orderInitializedEvent);
    }
}

File: order-service-kt/src/main/kotlin/com/intexsoft/webshop/orderservicekt/producer/impl/OrderEventProducerImpl.kt
package com.intexsoft.webshop.orderservicekt.producer.impl

import com.intexsoft.webshop.messagecommon.event.order.OrderInitializedEvent
import com.intexsoft.webshop.orderservicekt.log
import com.intexsoft.webshop.orderservicekt.producer.OrderEventProducer
import org.springframework.amqp.core.TopicExchange
import org.springframework.amqp.rabbit.core.RabbitTemplate
import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service

@Service
class OrderEventProducerImpl(
    private val rabbitTemplate: RabbitTemplate,
    private val eventExchange: TopicExchange
) : OrderEventProducer {
    @Value("\${rmq.event.order.routing-prefix}")
    private lateinit var routingPrefix: String

    @Value("\${rmq.event.order.routing-keys.order_initialized}")
    private lateinit var orderInitializedRoutingKey: String
    override fun produceOrderInitializedEvent(orderInitializedEvent: OrderInitializedEvent) {
        log.info("IN: produce ${orderInitializedEvent::class.simpleName} message. The event message = $orderInitializedEvent")
        val routing = routingPrefix + orderInitializedRoutingKey
        log.debug("Message is producing to exchange = ${eventExchange.name}, with routing = $routing")
        rabbitTemplate.convertAndSend(eventExchange.name, routing, orderInitializedEvent)
    }
}